code/go/0chain.net/smartcontract/dbs/event/user.go

```diff
@@ -142,23 +142,58 @@ func (edb *EventDb) updateUserPayedFees(users []User) error {
 }
 
 func mergeUpdateUserCollectedRewardsEvents() *eventsMergerImpl[User] {
-	return newEventsMerger[User](TagUpdateUserCollectedRewards, withUniqueEventOverwrite())
+	return newEventsMerger[User](TagUpdateUserCollectedRewards, withCollectedRewardsMerged())
+}
+
+func withCollectedRewardsMerged() eventMergeMiddleware {
+	return withEventMerge(func(a, b *User) (*User, error) {
+		a.CollectedReward += b.CollectedReward
+		return a, nil
+	})
 }
 
 func mergeUpdateUserTotalStakeEvents() *eventsMergerImpl[DelegatePoolLock] {
-	return newEventsMerger[DelegatePoolLock](TagLockStakePool, withUniqueEventOverwrite())
+	return newEventsMerger[DelegatePoolLock](TagLockStakePool, withTotalStakeMerged())
+}
+
+func withTotalStakeMerged() eventMergeMiddleware {
+	return withEventMerge(func(a, b *User) (*User, error) {
+		a.TotalStake += b.TotalStake
+		return a, nil
+	})
 }
 
 func mergeUpdateUserReadPoolTotalEvents() *eventsMergerImpl[ReadPoolLock] {
-	return newEventsMerger[ReadPoolLock](TagLockReadPool, withUniqueEventOverwrite())
+	return newEventsMerger[ReadPoolLock](TagLockReadPool, withReadPoolMerged())
+}
+
+func withReadPoolMerged() eventMergeMiddleware {
+	return withEventMerge(func(a, b *User) (*User, error) {
+		a.ReadPoolTotal += b.ReadPoolTotal
```

Comment:
`TagLockReadPool` emit event with data of `ReadPoolLock`, not `User`. Please also check others, and write unit tests to confirm that the changes work as expected.

Reply:
Fixed.

Reply:
Added unit tests